### PR TITLE
CascadeSelect - basic tests

### PR DESCRIPTION
**Pull request by christophs78**

### CODE REVIEW

File: src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect001.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        gpus.add(nVidia);
+        gpus.add(amd);
+    }
+
+    public void onItemSelect(SelectEvent event) {
+        // pause for 1 second to verify AJAX guards are working properly
+        TestUtils.wait(1000);
+
+        FacesMessage message = new FacesMessage(FacesMessage.SEVERITY_INFO, "Selected GPU", (String)event.getObject());
+        FacesContext.getCurrentInstance().addMessage(null, message);
```
Comment (melloware) [inline src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect001.java:91]: I have a handy `TestUtils.addMessage()` to reduce a lot of this boilerplate of creating INFO messages.

File: src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect002.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        gpus.add(nVidia);
+        gpus.add(amd);
+    }
+
+    public void onItemSelect(SelectEvent event) {
+        // pause for 1 second to verify AJAX guards are working properly
+        TestUtils.wait(1000);
+
+        FacesMessage message = new FacesMessage(FacesMessage.SEVERITY_INFO, "Selected GPU", Integer.toString((int)event.getObject()));
+        FacesContext.getCurrentInstance().addMessage(null, message);
```
Comment (melloware) [inline src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect002.java:88]: Same here

File: src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect002.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import javax.inject.Named;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+@Named
+@ViewScoped
+@Data
+public class CascadeSelect002 implements Serializable {
+
```
Comment (melloware) [inline src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect002.java:43]: maybe make an `AbstractCascadeSelect` class to reduce some of this copy and paste?

**Review by melloware — CHANGES_REQUESTED** [review]

File: src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect001.java (hunk 1 above)
Comment (christophs78) [reply src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect001.java]: good point!

File: src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect002.java (hunk 3 above)
Comment (christophs78) [reply src/main/java/org/primefaces/extensions/integrationtests/cascadeselect/CascadeSelect002.java:43]: Not sure there are redundancies we can refactor. CascadeSelect001 only uses labels for SelectItems. CascadeSelect001 adds values and labels to SelectItems.

**Comment by melloware** [comment]
OK do these tests need my NPE fix for them to pass?

**Comment by christophs78** [comment]
> OK do these tests need my NPE fix for them to pass?

No. But we need to fix https://github.com/primefaces/primefaces/issues/7082 for this tests to pass.

**Comment by melloware** [comment]
OK submitted fix for 7082 and it does handle converters.

**Comment by christophs78** [comment]
> OK submitted fix for 7082 and it does handle converters.

Thanks for working on this! IMO this was kind of a showstopper which made this new component almost unusable.

**Comment by melloware** [comment]
I totally agree I think Thomas will agree too and merge that fix tomorrow.

**Comment by christophs78** [comment]
I have to check this PR later this day. Probably there´s some issue with my  CascaseSelect-PF-Selenium-wrapper.

**Comment by melloware** [comment]
All tests passing and I just added tests for show/hide/enable/disable widget methods.

**Pull request merged**